src/Functions/CMakeLists.txt

```diff
@@ -51,71 +52,75 @@ option(STRIP_DEBUG_SYMBOLS_FUNCTIONS "Do not generate debugger info for ClickHou
 endif()
 
 if (TARGET ch_contrib::icu)
-    target_link_libraries (clickhouse_functions PRIVATE ch_contrib::icu)
+    list (APPEND PRIVATE_LIBS ch_contrib::icu)
 endif ()
 
 if (TARGET ch_contrib::fastops)
-    target_link_libraries (clickhouse_functions PRIVATE ch_contrib::fastops)
+    list (APPEND PRIVATE_LIBS ch_contrib::fastops)
 endif ()
 
 if (ENABLE_EXAMPLES)
     add_subdirectory(examples)
 endif ()
 
 if (TARGET ch_contrib::llvm)
-    target_link_libraries(clickhouse_functions PRIVATE ch_contrib::llvm)
+    list (APPEND PRIVATE_LIBS ch_contrib::llvm)
 endif ()
 
 if (TARGET ch_contrib::base-x)
-    target_link_libraries(clickhouse_functions PRIVATE ch_contrib::base-x)
+    list (APPEND PRIVATE_LIBS ch_contrib::base-x)
 endif()
 
 if (TARGET ch_contrib::base64)
-    target_link_libraries(clickhouse_functions PRIVATE ch_contrib::base64)
+    list (APPEND PRIVATE_LIBS ch_contrib::base64)
 endif()
 
-target_link_libraries(clickhouse_functions PRIVATE ch_contrib::lz4)
+list (APPEND PRIVATE_LIBS ch_contrib::lz4)
 
 if (ENABLE_NLP)
-    target_link_libraries(clickhouse_functions PRIVATE ch_contrib::cld2)
+    list (APPEND PRIVATE_LIBS ch_contrib::cld2)
 endif()
 
 if (TARGET ch_contrib::h3)
-    target_link_libraries (clickhouse_functions PRIVATE ch_contrib::h3)
+    list (APPEND PRIVATE_LIBS ch_contrib::h3)
 endif()
 
 if (TARGET ch_contrib::vectorscan)
-    target_link_libraries(clickhouse_functions PRIVATE ch_contrib::vectorscan)
+    list (APPEND PRIVATE_LIBS ch_contrib::vectorscan)
 endif()
 
 if (TARGET ch_contrib::simdjson)
-    target_link_libraries(clickhouse_functions PRIVATE ch_contrib::simdjson)
+    list (APPEND PRIVATE_LIBS ch_contrib::simdjson)
 endif()
 
 if (TARGET ch_contrib::rapidjson)
-    target_link_libraries(clickhouse_functions PRIVATE ch_contrib::rapidjson)
+    list (APPEND PRIVATE_LIBS ch_contrib::rapidjson)
 endif()
 
 add_subdirectory(GatherUtils)
-target_link_libraries(clickhouse_functions PRIVATE clickhouse_functions_gatherutils)
+list (APPEND PRIVATE_LIBS clickhouse_functions_gatherutils)
 
 add_subdirectory(URL)
-target_link_libraries(clickhouse_functions PRIVATE clickhouse_functions_url)
+list (APPEND PRIVATE_LIBS clickhouse_functions_url)
 
 add_subdirectory(array)
-target_link_libraries(clickhouse_functions PRIVATE clickhouse_functions_array)
+list (APPEND PRIVATE_LIBS clickhouse_functions_array)
 
 add_subdirectory(JSONPath)
+list (APPEND PRIVATE_LIBS clickhouse_functions_jsonpath)
 
 # Signed integer overflow on user-provided data inside boost::geometry - ignore.
 set_source_files_properties("pointInPolygon.cpp" PROPERTIES COMPILE_FLAGS -fno-sanitize=signed-integer-overflow)
 
 if (ENABLE_FUZZING)
     add_compile_definitions(FUZZING_MODE=1)
 endif ()
+
+target_link_libraries(clickhouse_functions_obj PUBLIC ${PUBLIC_LIBS} PRIVATE ${PRIVATE_LIBS})
+target_link_libraries(clickhouse_functions INTERFACE $<TARGET_OBJECTS:clickhouse_functions_obj> ${PUBLIC_LIBS} ${PRIVATE_LIBS})
```

Comment:
Can you please elaborate this change? INTERFACE library and using PRIVATE_LIBS/PUBLIC_LIBS to store the list of libraries.

Reply:
This is used to forward the linking information to the final binaries such as clickhouse / unit_tests_dbms, since such information are lost after we convert to OBJECT target.

Reply:
I see, indeed, thanks!

Reply:
Do I understand correctly that after this change we would not link `libclickhouse_functions.a` but just directly link all the functions like `log10.cpp.o` into `clickhouse` target?
If so, it is cool. (But I would be happy to see a comment here)

Reply:
You're right, `libclickhouse_functionsd.a` will be replaced by individual .o files.
The other method is linking with `-Wl,--whole-archive libclickhouse_functionsd.a  -Wl,--no-whole-archive`, personally I feel using OBJECT is better.